lib/acme/client.rb

```diff
@@ -88,24 +93,6 @@ def connection
     end
   end
 
-  def challenge_from_hash(arguments)
-    attributes = arguments.to_h
-    %w(type uri token).each do |key|
-      raise ArgumentError, "missing key: #{key}" unless attributes.key?(key)
-    end
-
-    case attributes.fetch('type')
-    when 'http-01'
-      Acme::Client::Resources::Challenges::HTTP01.new(self, attributes)
-    when 'dns-01'
-      Acme::Client::Resources::Challenges::DNS01.new(self, attributes)
-    when 'tls-sni-01'
-      Acme::Client::Resources::Challenges::TLSSNI01.new(self, attributes)
-    else
-      raise 'Unsupported resource type'
-    end
-  end
-
```

Comment:
Well, technically SemVer allows breaking API changes on minor releases while < 1.0, but it's up to you. I just don't see a sane way to support both APIs.